curator-framework/src/main/java/org/apache/curator/framework/imps/EnsembleTracker.java

```diff
@@ -209,10 +209,17 @@ private void processConfigData(byte[] data) throws Exception
         if (!properties.isEmpty())
         {
             QuorumMaj newConfig = new QuorumMaj(properties);
-            String connectionString = configToConnectionString(newConfig);
-            if (connectionString.trim().length() > 0)
+            String connectionString = configToConnectionString(newConfig).trim();
```

Comment:
Is it possible that the `connectionString` also has a `chroot` part and we append the old chroot last which is disordered?

Reply:
No. This `connectionString` is constructed from `QuorumMaj` which is a server side concept and has no notion of `chroot`.